db/migrate/20150713143226_add_unique_to_operatingsystems_title.rb

```diff
@@ -0,0 +1,17 @@
+class AddUniqueToOperatingsystemsTitle < ActiveRecord::Migration
+  def up
+    duplicities = Operatingsystem.all.group_by(&:title).values.keep_if { |ary| ary.length > 1}
+    duplicities.each do |systems|
+      systems.each do |os|
+        os.description = os.fullname
```

Comment:
Yes, it would override the description, because title is created from the description if it is present. If not, fullname is used as a description. I find the systems with duplicate title and then I set fullname as a description in double loop, because fullname is supposed to be unique. As a result, description and title are same as fullname and there are no duplicities in title.

We could also remove the title auto-generation from description and make users supply it.